hedgehog/src/Hedgehog/Internal/Gen.hs

```diff
@@ -305,9 +363,20 @@ class Monad m => MonadGen m where
   --
   freezeGen :: m a -> m (a, m a)
 
-instance Monad m => MonadGen (GenT m) where
```

Comment:
We remove the `MonadGen` instance for `GenT`, so the combinators in `Hedgehog.Gen` will now only work on `Gen` or transformers including `Gen` as the base monad.

Reply:
In the 99% case, users will just use `Gen` by virtue of the type of `forAll` and not worry about any of this.